UM/Settings/MachineManager.py

```diff
@@ -109,8 +156,11 @@ def removeMachineInstance(self, instance):
         self._machine_instances.remove(instance)
         instance.nameChanged.disconnect(self._onInstanceNameChanged)
 
+        file_name = urllib.parse.quote_plus(instance.getName())
         try:
-            path = Resources.getStoragePath(Resources.MachineInstances, urllib.parse.quote_plus(instance.getName()) + ".cfg")
+            path = Resources.getStoragePath(Resources.MachineInstances, file_name + ".cfg")
+            os.remove(path)
+            path = Resources.getStoragePath(Resources.MachineInstanceProfiles, file_name + ".cfg")
```

Comment:
There could be more than one profile associated with one machine instance though?